node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminImpl.java

```diff
@@ -139,11 +140,20 @@ public void updateMetrics(boolean isSuspended) {
         if (!isSuspended) {
             containerCount.sample(numContainers);
             ProcMeminfo meminfo = procMeminfoReader.read();
-            metrics.declareGauge("mem.system.overhead", new Dimensions(Map.of("containers", Long.toString(numContainers))))
-                   .sample(meminfo.memTotalBytes() - meminfo.memAvailableBytes() - totalContainerMemoryBytes);
+            updateMemoryOverheadMetric(numContainers, meminfo.memTotalBytes() - meminfo.memAvailableBytes() - totalContainerMemoryBytes);
         }
     }
 
+    private void updateMemoryOverheadMetric(int numContainers, double memoryOverhead) {
+        final String name = "mem.system.overhead";
+        Dimensions dimensions = new Dimensions(Map.of("containers", Integer.toString(numContainers)));
+        metrics.declareGauge(Metrics.APPLICATION_HOST, name, dimensions, Metrics.DimensionType.DEFAULT)
+               .sample(memoryOverhead);
+        if (previousMemoryOverheadDimensions != null && !previousMemoryOverheadDimensions.equals(dimensions))
```

Comment:
```suggestion
        if (!dimensions.equals(previousMemoryOverheadDimensions))
```

Reply:
This would be wrong if previousMemoryOverheadDimensions is null.

Reply:
Doh! :facepalm: